composer/callbacks/speed_monitor.py

```diff
@@ -63,106 +63,106 @@ class SpeedMonitor(Callback):
     """
 
     def __init__(self, window_size: int = 100):
-        super().__init__()
-        self.train_examples_per_epoch = 0
-
-        # log the total wall clock time that this program has been running for
-        self.wall_clock_total = 0.0
+        # Track the epoch num samples and wct to compute throughput over the entire epoch
+        self.epoch_start_num_samples = Time.from_sample(0)
+        self.epoch_start_wct = datetime.timedelta(0)
+
+        # Track the batch num samples and wct to compute throughput over a window of batches
+        self.batch_start_num_samples = Time.from_sample(0)
+        self.batch_start_wct = datetime.timedelta(0)
+        self.rolling_batch_times_and_num_samples: Deque[Tuple[datetime.timedelta,
+                                                              Time[int]]] = deque(maxlen=window_size)
+        self.window_size = window_size
 
-        # log the total wall clock_time that has been spent in training
-        self.wall_clock_train = 0.0
+        # To optimize performance for the the batch wct calculations, keep a running sum over the entire deque
+        self.running_wct = datetime.timedelta(0)
```

Comment:
we use `running` here but `total` for evaluation.. maybe just use `total`?

Reply:
I'll rename `running` to `window` to be more specific. The evaluation time is across all batches and epochs, not just the current window.